tensornetwork/backends/jax/jitted_functions.py

```diff
@@ -721,6 +726,65 @@ def gmres_krylov(A_mv: Callable, A_args: Sequence, n_kry: int,
     # The 'x' input for the carry call. Each iteration will receive an ascending
     # loop index (from the jnp.arange) along with the constant data
     # in gmres_constants.
+
+    def gmres_krylov_work(gmres_carry: GmresCarryType) -> GmresCarryType:
+      """
+      Performs a single iteration of gmres_krylov. See that function for a more
+      detailed description.
+  
+      Args:
+        gmres_carry: The gmres_carry from gmres_krylov.
+      Returns:
+        gmres_carry: The updated gmres_carry.
+      """
+      gmres_variables, gmres_constants = gmres_carry
+      k, V, R, beta_vec, err, givens = gmres_variables
+      tol, A_mv, A_args, b_norm, _ = gmres_constants
+
+      V, H = kth_arnoldi_step(k, A_mv, A_args, V, R, tol, precision)
+      R_col, givens = apply_givens_rotation(H[:, k], givens, k)
+      R = jax.ops.index_update(R, jax.ops.index[:, k], R_col[:])
+
+      # Update the residual vector.
+      cs, sn = givens[:, k] * beta_vec[k]
+      beta_vec = jax.ops.index_update(beta_vec, jax.ops.index[k], cs)
+      beta_vec = jax.ops.index_update(beta_vec, jax.ops.index[k + 1], sn)
+      err = jnp.abs(sn) / b_norm
+      gmres_variables = (k + 1, V, R, beta_vec, err, givens)
+      return (gmres_variables, gmres_constants)
+
+    def gmres_krylov_loop_condition(gmres_carry: GmresCarryType) -> bool:
+      """
+      This function dictates whether the main GMRES while loop will proceed.
```

Comment:
It's so difficult to know what you're actually changing when you have these huge refactors randomly.